lib/nav.py

```diff
@@ -36,3 +56,116 @@ def force_dir(dir, n=1):
 			if ACTIONS[OPS.MOVE][dir]():
 				break
 			ACTIONS[OPS.DIG][dir]()
+
+# move n blocks
+def dir(dir, n=1):
+	for i in range(n):
+		fuel.fuel()
+		if not ACTIONS[OPS.MOVE][dir]():
+			return False
+	return True
+
+# turn a by a given direction
+def turn(turn_dir):
+	ACTIONS[OPS.TURN][turn_dir]()
+
+class Navigator:
+	def __init__(self, location=[0,0,0], direction=CARDINALS.NORTH):
```

Comment:
There's a tricky bug here with `location=[0,0,0]`, as that is a mutable value

What you actually need to do is:
```
def __init__(..., location=None, ...):
  self.location = location if location is not None else []
```

See this page for more details on the bug: https://www.geeksforgeeks.org/default-arguments-in-python/

Specifically search for the "Using mutable objects as default argument values in python" heading

Reply:
Done